File: SDKs/apideck.com/connector/8.70.2/python/src/sdk/models/operations/connectorsone.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class ConnectorsOnePathParams:
    id: str = field(default=None, metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass
class ConnectorsOneHeaders:
    x_apideck_app_id: str = field(default=None, metadata={'header': { 'field_name': 'x-apideck-app-id', 'style': 'simple', 'explode': False }})
    

@dataclass
class ConnectorsOneSecurity:
    api_key: shared.SchemeAPIKey = field(default=None, metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    

@dataclass
class ConnectorsOneRequest:
    path_params: ConnectorsOnePathParams = field(default=None)
    headers: ConnectorsOneHeaders = field(default=None)
    security: ConnectorsOneSecurity = field(default=None)
    

@dataclass
class ConnectorsOneResponse:
    content_type: str = field(default=None)
    get_connector_response: Optional[shared.GetConnectorResponse] = field(default=None)
    not_found_response: Optional[shared.NotFoundResponse] = field(default=None)
    payment_required_response: Optional[shared.PaymentRequiredResponse] = field(default=None)
    status_code: int = field(default=None)
    unauthorized_response: Optional[shared.UnauthorizedResponse] = field(default=None)
    unexpected_error_response: Optional[shared.UnexpectedErrorResponse] = field(default=None)
    

File: SDKs/apideck.com/connector/8.70.2/typescript/src/sdk/models/operations/connectorsone.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class ConnectorsOnePathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class ConnectorsOneHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=x-apideck-app-id" })
  xApideckAppId: string;
}


export class ConnectorsOneSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKey: shared.SchemeApiKey;
}


export class ConnectorsOneRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: ConnectorsOnePathParams;

  @Metadata()
  headers: ConnectorsOneHeaders;

  @Metadata()
  security: ConnectorsOneSecurity;
}


export class ConnectorsOneResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  getConnectorResponse?: shared.GetConnectorResponse;

  @Metadata()
  notFoundResponse?: shared.NotFoundResponse;

  @Metadata()
  paymentRequiredResponse?: shared.PaymentRequiredResponse;

  @Metadata()
  statusCode: number;

  @Metadata()
  unauthorizedResponse?: shared.UnauthorizedResponse;

  @Metadata()
  unexpectedErrorResponse?: shared.UnexpectedErrorResponse;
}
